src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.ComposeCache.targets

```diff
@@ -35,14 +145,14 @@ Copyright (c) .NET Foundation. All rights reserved.
     Prepare the prerequisites for ComposeCache.
     ============================================================
     -->
+  
   <Target Name="PrepareForComposeCache">
 
     <PropertyGroup>
       <PreserveCacheLayout Condition="'$(PreserveCacheLayout)' == ''">true</PreserveCacheLayout>
       <SkipOptimization Condition="'$(RuntimeIdentifier)' == ''">true</SkipOptimization>
```

Comment:
The absence of an RID indicates that we can not do any optimization, so this should be an unconditional check. Anyway this condition will not be triggered in regular workflow